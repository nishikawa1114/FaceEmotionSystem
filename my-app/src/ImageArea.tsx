import React from 'react';
import './index.css';
import { ImageInfo } from './ImageInfo';

interface Image {
    id: number;
    url: string;
}

interface BoardProps {
    images: Array<Image>;
    checkedImages: Array<boolean>
    onClick: (i: number) => void;
}

export class ImageArea extends React.Component<BoardProps>  {

    // それぞれの画像を表示する
    private renderImageInfo(i: number) {
        return (
            <ImageInfo
                image={this.props.images[i]}
                onClick={() => this.props.onClick(i)}
                checked={this.props.checkedImages[i]}
            />
        );
    }

    public render() {
        const length = this.props.images.length;
        // 全ての画像を表示する
        return (
            <div className="image_area">
                {
                    Array(length).fill(this.props.images).map((value, i: number) => {
                        return (
                            this.renderImageInfo(i)
                        );
                    })
                }
            </div>
        )
    }
}