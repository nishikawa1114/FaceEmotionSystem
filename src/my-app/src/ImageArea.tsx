import React from 'react';
import './index.css';
import { ImageUrl } from './types';
import { Checkbox, GridList, GridListTile, GridListTileBar, IconButton, ListSubheader } from '@material-ui/core';

interface BoardProps {
    images: Array<ImageUrl>;
    checkedImages: Array<boolean>
    onClick: (i: number) => void;
}

// ホーム画面内に画像リストを表示するコンポーネント
export class ImageArea extends React.Component<BoardProps>  {

    // URLから日付を返す
    getDate = (str: string) => {
        const strDate = String(str.match(/\d{4}\/\d{2}\/\d{2}/));
        return strDate;
    }

    // URLからユーザー名を返す
    getName = (str: string) => {
        const temp: string = String(str.split('images/').pop());
        const name: string = String(temp.split('/').shift());
        return name;
    }

    public render() {
        // 全ての画像を表示する
        const length = this.props.images.length;
        return (
            <div className="image_area">
                <GridList cols={4} key="image_list">
                    {
                        Array(length).fill(this.props.images).map((value, i: number) => (
                            <GridListTile key={i}>
                                <img src={value[i].url} onClick={() => this.props.onClick(i)} className="image" key={i} />
                                {/* 画像のユーザー名と日付を表示 */}
                                <GridListTileBar
                                    title={this.getName(value[i].url)}
                                    subtitle={this.getDate(value[i].url)}
                                    actionIcon={
                                        <Checkbox
                                            edge="end"
                                            color="secondary"
                                            value={this.props.images[i].url}
                                            checked={this.props.checkedImages[i]}
                                            className="image_checkbox"
                                            key={i}
                                        />
                                    }
                                />
                            </GridListTile>
                        ))
                    }
                </GridList>
            </div>
        )
    }
}