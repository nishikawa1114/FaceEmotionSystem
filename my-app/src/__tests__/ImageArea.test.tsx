import { ImageArea } from "../ImageArea";
import { create } from "react-test-renderer";
import React from "react";

interface Image {
    id: number;
    url: string;
}

const images: Array<Image> = [
    {
        id: 1,
        url: "https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D",
    },
    {
        id: 2,
        url: "https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D",
    }
]

const checkedImages: Array<boolean> = [true, false];

    it("snapshot test", () => {
        const tree = create(
            <ImageArea
                images={images}
                checkedImages={checkedImages}
                onClick={() => {}}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();

    })