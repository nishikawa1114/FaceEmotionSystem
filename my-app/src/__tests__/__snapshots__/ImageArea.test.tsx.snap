// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="image_area"
>
  <div
    className="image_info"
  >
    <img
      className="image"
      onClick={[Function]}
      src="https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D"
    />
    <form>
      <input
        checked={true}
        className="image_checkbox"
        type="checkbox"
        value="https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D"
      />
    </form>
    steve
    <br />
    2020/10/15
  </div>
  <div
    className="image_info"
  >
    <img
      className="image"
      onClick={[Function]}
      src="https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D"
    />
    <form>
      <input
        checked={false}
        className="image_checkbox"
        type="checkbox"
        value="https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D"
      />
    </form>
    steve
    <br />
    2020/10/15
  </div>
</div>
`;
